stdlib/benchmarks/collections/bench_string.mojo

```diff
@@ -0,0 +1,323 @@
+# Benchmark string replace
+# ===----------------------------------------------------------------------===#
+@parameter
+fn bench_string_replace[
+    length: UInt = 0,
+    filename: String = "UN charter EN",
+    old: String = "a",
+    new: String = "A",
+](inout b: Bencher) raises:
+    var items = make_string[length, filename + ".txt"]()
+
+    @always_inline
+    @parameter
+    fn call_fn() raises:
+        var res = items.replace(old, new)
+        keep(res._buffer.data)
+
+    b.iter[call_fn]()
+    keep(bool(items))
+
+
+# ===----------------------------------------------------------------------===#
+# Benchmark string _is_valid_utf8
+# ===----------------------------------------------------------------------===#
+@parameter
+fn bench_string_is_valid_utf8[
+    length: UInt = 0, filename: String = "UN charter EN"
+](inout b: Bencher) raises:
+    var items = make_string[length, filename + ".html"]()
+
+    @always_inline
+    @parameter
+    fn call_fn() raises:
+        var res = _is_valid_utf8(items.unsafe_ptr(), length)
+        keep(res)
+
+    b.iter[call_fn]()
+    keep(bool(items))
+
+
+# ===----------------------------------------------------------------------===#
+# Benchmark Main
+# ===----------------------------------------------------------------------===#
+def main():
+    seed()
+    var m = Bench(BenchConfig(num_repetitions=1))
+    m.bench_function[bench_string_init](BenchId("bench_string_init"))
+    alias filenames = (
+        "UN charter EN",
+        "UN charter ES",
+        "UN charter AR",
+        "UN charter RU",
+        "UN charter zh-CN",
+    )
+    alias old_new_chars = (
+        ("a", "A"),
+        ("ó", "Ó"),
+        ("ل", "ل"),
+        ("и", "И"),
+        ("一", "一"),
+    )
+    alias lengths = (
+        10,
+        20,
+        30,
+        40,
+        50,
+        60,
+        70,
+        80,
+        90,
+        100,
+        200,
+        300,
+        400,
+        500,
+        600,
+        700,
+        800,
+        900,
+        1000,
+        2000,
+        3000,
+        4000,
+        5000,
+        6000,
+        7000,
+        8000,
+        9000,
+        10_000,
+        20_000,
+        30_000,
+        40_000,
+        50_000,
+        60_000,
+        70_000,
+        80_000,
+        90_000,
+        100_000,
+        200_000,
+        300_000,
+        400_000,
+        500_000,
+        600_000,
+        700_000,
+        800_000,
+        900_000,
+        1_000_000,
+    )
+
+    @parameter
+    for i in range(len(lengths)):
+        alias length = lengths.get[i, Int]()
+
+        @parameter
+        for j in range(len(filenames)):
+            alias fname = filenames.get[j, StringLiteral]()
+            alias chars = old_new_chars.get[j, Tuple[String, String]]()
+            alias old = chars.get[0, String]()
+            alias new = chars.get[1, String]()
+            m.bench_function[bench_string_count[length, fname, old]](
```

Comment:
FYI this is crashing internally:

```
stdlib/benchmarks/collections/bench_string.mojo.test.runfiles/_main/open-source/mojo/stdlib/benchmarks/collections/bench_string.mojo -t
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:113:4: error: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:96:4: note: function instantiation failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:108:57: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:68:4: note: function instantiation failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:84:18: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/_startup.mojo:105:8: note: function instantiation failed
/mnt/engflow/worker/work/2/exec/bazel-out/k8-opt-release/bin/open-source/mojo/stdlib/benchmarks/collections/bench_string.mojo.test.runfiles/_main/open-source/mojo/stdlib/benchmarks/collections/bench_string.mojo:299:69: note: call expansion failed
            m.bench_function[bench_string_count[length, fname, old]](
                                                                    ^
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/tuple.mojo:181:8: note: function instantiation failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/tuple.mojo:191:25: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/rebind.mojo:70:70: note: call expansion failed
/mnt/engflow/worker/work/0/exec/open-source/mojo/stdlib/stdlib/builtin/rebind.mojo:42:52: note: error: rebind input type '!kgen.pointer<struct<(!kgen.pack<[[string, {"__copyinit__" : (!kgen.pointer<string> init_self, !kgen.pointer<string> borrow_in_mem) -> !kgen.none = @"stdlib::builtin::string_literal::StringLiteral::__copyinit__(::StringLiteral=&,::StringLiteral)_thunk", "__del__" : (!kgen.pointer<string> owned_in_mem) -> !kgen.none = @"stdlib::builtin::string_literal::StringLiteral::__del__(::StringLiteral)_thunk", "__moveinit__" : (!kgen.pointer<string> init_self, !kgen.pointer<string> owned_in_mem) -> !kgen.none = @"stdlib::builtin::string_literal::StringLiteral::__moveinit__(::StringLiteral=&,::StringLiteral)_thunk"}], [string, {"__copyinit__" : (!kgen.pointer<string> init_self, !kgen.pointer<string> borrow_in_mem) -> !kgen.none = @"stdlib::builtin::string_literal::StringLiteral::__copyinit__(::StringLiteral=&,::StringLiteral)_thunk", "__del__" : (!kgen.pointer<string> owned_in_mem) -> !kgen.none = @"stdlib::builtin::string_literal::StringLiteral::__del__(::StringLiteral)_thunk", "__moveinit__" : (!kgen.pointer<string> init_self, !kgen.pointer<string> owned_in_mem) -> !kgen.none = @"stdlib::builtin::string_literal::StringLiteral::__moveinit__(::StringLiteral=&,::StringLiteral)_thunk"}]]>) memoryOnly>>' does not match result type '!kgen.pointer<struct<(!kgen.pack<[[struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>, {"__copyinit__" : (!kgen.pointer<struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>> init_self, !kgen.pointer<struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>> borrow_in_mem) -> !kgen.none = @"stdlib::collections::string::String::__copyinit__(stdlib::collections::string::String=&,stdlib::collections::string::String)", "__del__" : (!kgen.pointer<struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>> owned_in_mem) -> !kgen.none = @"stdlib::collections::string::String::__del__(stdlib::collections::string::String)", "__moveinit__" : (!kgen.pointer<struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>> init_self, !kgen.pointer<struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>> owned_in_mem) -> !kgen.none = @"stdlib::collections::string::String::__moveinit__(stdlib::collections::string::String=&,stdlib::collections::string::String)"}], [struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>, {"__copyinit__" : (!kgen.pointer<struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>> init_self, !kgen.pointer<struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>> borrow_in_mem) -> !kgen.none = @"stdlib::collections::string::String::__copyinit__(stdlib::collections::string::String=&,stdlib::collections::string::String)", "__del__" : (!kgen.pointer<struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>> owned_in_mem) -> !kgen.none = @"stdlib::collections::string::String::__del__(stdlib::collections::string::String)", "__moveinit__" : (!kgen.pointer<struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>> init_self, !kgen.pointer<struct<(struct<(pointer<none>, index, index) memoryOnly>) memoryOnly>> owned_in_mem) -> !kgen.none = @"stdlib::collections::string::String::__moveinit__(stdlib::collections::string::String=&,stdlib::collections::string::String)"}]]>) memoryOnly>>'
mojo: error: failed to run the pass manager
```

Do you mind trying to minimize and repro this?  It's in the compile time interpreter, likely due to the use of the `Tuple` get stuff for `old` and `new`.  I think if you split it into a standalone mojo file without all the impl details, you should be able to get to a minimal crasher/reproducer and hopefully find a workaround - otherwise, file a bug, but I'd hope we could find a workaround in this case.  Would love to land the benchmarks sooner rather than later as you're actively working in this area and we could then more easily quantify the efforts/perf gains.